### PR TITLE
Feature/design system

### PR DESCRIPTION
**Pull request by SolMendiola**
# :star: Feature

---

#### :pencil2: Description:

Added design system

---

#### iOS

Screen Shot 2022-10-28 at 16 22 23

Simulator Screen Shot - iPhone 13 - 2022-10-28 at 16 23 48

### CODE REVIEW

File: lib/ui/common/input_text.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          suffixIcon: error != null
+              ? IconButton(onPressed: actionSecondIcon, icon: Icon(secondIcon))
+              : Icon(Icons.error),
+          labelStyle:
+              context.theme.textStyles.bodyLarge?.copyWith(color: Colors.black),
+          hintStyle:
+              context.theme.textStyles.bodyLarge?.copyWith(color: Colors.black),
+        ),
+        style:
+            context.theme.textStyles.bodyLarge?.copyWith(color: Colors.black),
```
Comment (mirland) [inline lib/ui/common/input_text.dart:57]: you cannot use a fixed color, it depends on the theme

File: lib/ui/common/app_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  Widget build(BuildContext context) => Container(
+        width: 100,
+        height: 100,
+        child: AlertDialog(
+          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(28.r)),
+          backgroundColor: context.theme.colors.primary.shade100,
+          title: Column(
+            children: [
+              Icon(Icons.mobile_friendly_outlined),
+              SizedBox(height: 20,),
```
Comment (mirland) [inline lib/ui/common/app_dialog.dart:35]:
```suggestion
              SizedBox(height: 20),
```

File: lib/ui/theme/app_colors.dart [hunk 3]
```diff
@@ -23,31 +31,66 @@ class AppColors extends ColorScheme {
...
-        secondary: generateMaterialColor(color: Color(0xffffd81d)),
-        onSecondary: Colors.black,
-        error: generateMaterialColor(color: Color(0xffF4642C)),
-        onError: Colors.black,
+        primary: MaterialColor(0xff042B59, {
+          400: Color(0xff2C649B),
+          300: Color(0xff5798CD),
+          200: Color(0xff93C9EE),
+          100: Color(0xffC7E5F6),
+        }),
```
Comment (mirland) [inline lib/ui/theme/app_colors.dart:60]: move to one variable?

File: lib/ui/theme/app_colors.dart [hunk 4]
```diff
@@ -23,31 +31,66 @@ class AppColors extends ColorScheme {
...
+          },
+        ),
+        onSecondary: Colors.white,
+        error: MaterialColor(
+          0xffFF5160,
+          {
+            200: Color(0xffFF7D7C),
+            300: Color(0xffFFA096),
+          },
+        ),
```
Comment (mirland) [inline lib/ui/theme/app_colors.dart:76]: same?

File: lib/ui/tips/tips_screen.dart [hunk 5]
```diff
@@ -16,18 +22,55 @@ class TipsScreen extends StatelessWidget {
...
-        return Scaffold(
-          body: ListView.builder(
+        return Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            RaisedButton(
+              onPressed: () => showDialog(
+                context: context,
+                builder: (_) => AppDialog(
+                  title: 'titulo',
```
Comment (mirland) [inline lib/ui/tips/tips_screen.dart:38]: english?

File: lib/ui/tips/tips_screen.dart [hunk 6]
```diff
@@ -16,18 +22,55 @@ class TipsScreen extends StatelessWidget {
 }
 
 class _TipContentScreen extends StatelessWidget {
```
Comment (mirland) [inline lib/ui/tips/tips_screen.dart:24]: I would create a catalog screen

**Review by mirland — CHANGES_REQUESTED** [review]

File: lib/ui/catalog/catalogScreen.dart [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
```
Comment (mirland) [inline lib/ui/catalog/catalogScreen.dart:1]: the file name is wrong

File: lib/ui/catalog/catalogScreen.dart [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          ElevatedButton(
+            onPressed: () => showDialog(
+              context: context,
+              builder: (_) => AppDialog(
+                title: 'title',
+                negativeOption: 'negative',
+                positiveOption: 'positive',
+                contentText: 'context',
```
Comment (mirland) [inline lib/ui/catalog/catalogScreen.dart:26]:
```suggestion
                contentText: 'content',
```

File: lib/ui/common/app_dialog.dart [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  final String? negativeOption;
+  final String positiveOption;
+  final VoidCallback negativeAction;
+  final VoidCallback positiveAction;
+
+  const AppDialog({
+    Key? key,
+    required this.title,
+    required this.contentText,
+    required this.negativeOption,
```
Comment (mirland) [inline lib/ui/common/app_dialog.dart:18]: it shouldnt be required

File: lib/ui/common/app_dialog.dart [hunk 10]
```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_template/ui/extensions/context_extensions.dart';
+import 'package:flutter_template/ui/theme/app_theme.dart';
+
+class AppDialog extends StatelessWidget {
+  final String title;
+  final String contentText;
+  final String? negativeOption;
+  final String positiveOption;
```
Comment (mirland) [inline lib/ui/common/app_dialog.dart:10]: shouldnt be option1 and option two?

File: lib/ui/common/input_text.dart [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  final String? textHint;
+  final void Function()? actionSecondIcon;
+  final Color? backgroundColor;
+  final String? error;
+
+  const InputText({
+    Key? key,
+    required this.controller,
+    this.height,
+    this.width,
```
Comment (mirland) [inline lib/ui/common/input_text.dart:21]: Are you using it?

File: lib/ui/common/input_text.dart [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    this.trailingIcon,
+    this.textHint,
+    this.actionSecondIcon,
+    this.backgroundColor,
+    this.error,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => TextField(
+        onChanged: (_) {},
```
Comment (mirland) [inline lib/ui/common/input_text.dart:32]: calback?

File: lib/ui/common/search.dart [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  @override
+  Widget build(BuildContext context) => Container(
+        alignment: Alignment.centerRight,
+        child: Container(
+          child: InputText(
+            leadingIcon: Icons.search,
+            trailingIcon: Icons.highlight_off,
+            controller: controller,
+            textHint: 'Search',
+            // string
```
Comment (mirland) [inline lib/ui/common/search.dart:21]: check it

File: lib/ui/tips/tips_screen.dart [hunk 14]
```diff
@@ -16,18 +17,22 @@ class TipsScreen extends StatelessWidget {
...
+        return CatalogScreen(
+          controller: controller,
+        );
+        /*ListView.builder(
             shrinkWrap: true,
             itemCount: 11,
             itemBuilder: (BuildContext context, int index) =>
                 Image.network(state.tips[index].imageUrl),
           ),
-        );
```
Comment (mirland) [inline lib/ui/tips/tips_screen.dart:30]: rollback

File: lib/ui/common/app_dialog.dart [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+    required this.negativeAction,
+    required this.positiveAction,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => Container(
+        constraints: BoxConstraints(
+          minHeight: 100.h,
+          maxHeight: 400.w,
+        ),
```
Comment (f7deleon) [inline lib/ui/common/app_dialog.dart:29]: what happens if the content is greater than 400.w?

File: lib/ui/common/app_dialog.dart [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+          ),
+          content: Text(
+            contentText,
+            style: context.theme.textTheme.bodyMedium,
+          ),
+          actions: [
+            TextButton(
+              onPressed: negativeAction,
+              child: Text(
+                negativeOption!,
```
Comment (f7deleon) [inline lib/ui/common/app_dialog.dart:53]: i think that if you required this you should make it non nullable

File: lib/ui/common/fab.dart [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+      {Key? key,
+      required this.state,
+      this.iconSelected,
+      required this.iconNotSelected})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return FloatingActionButton(
+      onPressed: () {},
```
Comment (f7deleon) [inline lib/ui/common/fab.dart:24]: what happends if its pressed?

File: lib/ui/common/search.dart [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+
+  @override
+  Widget build(BuildContext context) => Container(
+        alignment: Alignment.centerRight,
+        child: Container(
+          child: InputText(
+            leadingIcon: Icons.search,
+            trailingIcon: Icons.highlight_off,
+            controller: controller,
+            textHint: 'Search',
```
Comment (f7deleon) [inline lib/ui/common/search.dart:20]: localize this string

File: lib/ui/common/search.dart [hunk 19]
```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_template/ui/common/input_text.dart';
+
+class Search extends StatelessWidget {
```
Comment (f7deleon) [inline lib/ui/common/search.dart:4]: i think that search is maybe too general what about SearchBar or something like that?

File: lib/ui/theme/app_colors.dart [hunk 20]
```diff
@@ -23,31 +62,40 @@ class AppColors extends ColorScheme {
...
+        primary: _primaryColors,
+        onPrimary: _waterColor,
+        secondary: _secondaryColors,
+        onSecondary: Colors.white,
+        error: _errorColors,
+        onError: Colors.white,
         onBackground: Colors.black,
-        surface: generateMaterialColor(color: Color(0xFFE6E9EC)),
-        onSurface: Colors.black,
+        surface: generateMaterialColor(color: Color(0xffE0F4FF)),
```
Comment (f7deleon) [inline lib/ui/theme/app_colors.dart:93]: this is ok?

**Review by f7deleon — CHANGES_REQUESTED** [review]

File: lib/ui/catalog/catalog_screen.dart [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+    controllerSearchBar.dispose();
+    controllerAppButton.dispose();
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
```
Comment (mirland) [inline lib/ui/catalog/catalog_screen.dart:27]: Do you need it?
```suggestion
```

File: lib/ui/common/app_dialog.dart [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+            TextButton(
+              onPressed: actionOne,
+              child: Text(
+                optionOne,
+                style: context.theme.textTheme.labelLarge!
+                    .copyWith(color: context.theme.colors.tertiary),
+                textAlign: TextAlign.right,
+              ),
+            ),
+            if (optionTwo is String)
```
Comment (mirland) [inline lib/ui/common/app_dialog.dart:59]: why? check if it not null

File: lib/ui/common/fab.dart [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+  final IconData? iconSelected;
+  final IconData iconNotSelected;
+  final VoidCallback? action;
+
+  const Fab(
+      {Key? key,
+      required this.state,
+      this.iconSelected,
+      required this.iconNotSelected,
+      this.action})
```
Comment (mirland) [inline lib/ui/common/fab.dart:20]: comma

File: lib/ui/common/input_text.dart [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+    this.textHint,
+    this.actionSecondIcon,
+    this.backgroundColor,
+    this.error,
+    this.action,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => TextField(
+        onChanged: (word) => action!(word),
```
Comment (mirland) [inline lib/ui/common/input_text.dart:30]: action could be null, fix it

**Review by f7deleon — APPROVED** [review]

File: lib/l10n/intl_en.arb [hunk 25]
```diff
@@ -19,5 +19,6 @@
   "mail": "Mail",
   "password": "Password",
   "home": "Home",
-  "xmartlabs_projects": "Xmartlabs' projects"
+  "xmartlabs_projects": "Xmartlabs' projects",
+  "search": "Search"
 }
```
Comment (mirland) [inline lib/l10n/intl_en.arb:24]:
```suggestion
}

```

**Review by mirland — APPROVED** [review]

**Pull request merged**